services/pull/merge.go

```diff
@@ -30,8 +31,68 @@ import (
 	"code.gitea.io/gitea/modules/timeutil"
 	asymkey_service "code.gitea.io/gitea/services/asymkey"
 	issue_service "code.gitea.io/gitea/services/issue"
+
+	"github.com/unknwon/com"
 )
 
+// GetDefaultMergeMessage returns default message used when merging pull request
+func GetDefaultMergeMessage(baseGitRepo *git.Repository, pr *models.PullRequest, mergeStyle repo_model.MergeStyle) string {
+	if err := pr.LoadHeadRepo(); err != nil {
+		log.Error("LoadHeadRepo[%d]: %v", pr.HeadRepoID, err)
+		return ""
+	}
+	if err := pr.LoadIssue(); err != nil {
+		log.Error("Cannot load issue %d for PR id %d: Error: %v", pr.IssueID, pr.ID, err)
+		return ""
+	}
+	if err := pr.LoadBaseRepo(); err != nil {
+		log.Error("LoadBaseRepo: %v", err)
+		return ""
+	}
+
+	issueReference := "#"
+	if pr.BaseRepo.UnitEnabled(unit.TypeExternalTracker) {
+		issueReference = "!"
+	}
+
+	if mergeStyle != "" {
+		templateFilepath := fmt.Sprintf(".gitea/default_merge_message/%s_TEMPLATE.md", strings.ToUpper(string(mergeStyle)))
+		commit, err := baseGitRepo.GetBranchCommit(pr.BaseRepo.DefaultBranch)
+		if err != nil {
+			log.Error("GetBranchCommit: %v", err)
+			return ""
+		}
+		templateContent, ok := commit.GetFileContent(templateFilepath)
+		if ok {
+			return com.Expand(templateContent,
+				map[string]string{
+					"BaseRepoOwner":     pr.BaseRepo.OwnerName,
+					"BaseRepoName":      pr.BaseRepo.Name,
+					"BaseBranch":        pr.BaseBranch,
+					"HeadRepoOwner":     pr.HeadRepo.OwnerName,
+					"HeadRepoName":      pr.HeadRepo.Name,
+					"HeadBranch":        pr.HeadBranch,
+					"PullRequestTitle":  pr.Issue.Title,
+					"PullRequestBody":   pr.Issue.Content,
+					"PullRequestPoster": pr.Issue.Poster.Name,
+					"PullRequestIndex":  strconv.FormatInt(pr.Index, 10),
+					"IssueReference":    issueReference,
+				})
```

Comment:
Is there an efficient way to get the issues that will be closed by a PR?
Because I think that would be useful here as well to insert.